### ISSUE TITLE
Detect if an applet has been removed from the panel

### ISSUE THREAD

**Issue by fossfreedom** [solus-project/budgie-desktop-examples#7]
Looking to create a test python applet which will invoke an action when the applet is removed from the panel.

Any thoughts how to-do this?

**Comment by ikeydoherty** [comment]
Hm. Can I ask the use case?

**Comment by fossfreedom** [comment]
sure.

Trying to find a method to remove a custom keybinding - when the applet is active its easy enough to add the keybinding via the applet init.  If the applet is subsequently removed, that keybinding should be removed.

**Comment by ikeydoherty** [comment]
Oh so you're trying to do it during the lifecycle of the applet, not detect **other** applets? You could hook up the `destroy` signal and do like a `GLib.message()` to verify its being invoked?

**Comment by fossfreedom** [comment]
correct - applet looking after itself.

oddly it doesnt appear that destroy is being invoked on either the applet nor plugin objects.

**Comment by yursan9** [comment]
I agree with @fossfreedom .Can you add method to detect if the applet is deleted from panel?
